Assets/Scripts/PlayerController.cs

```diff
@@ -21,31 +23,79 @@ override protected bool ContainsEnemy(Tile tile)
         return tile.HasNPC();
     }
 
+    private Tile GetMouseTile()
+    {
+        Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+        RaycastHit hit;
+        // Ignore a click on empty space
+        if (!Physics.Raycast(ray, out hit)) return null;
+
+        if (hit.collider.tag == "Tile")
+        {
+            Tile mouseTile = hit.collider.GetComponent<Tile>();
+            return mouseTile;
+        }
+        return null;
+    }
+
+    void ClearMouseHover()
+    {
+        foreach (GameObject line in GameObject.FindGameObjectsWithTag("LineTag"))
+        {
+            Destroy(line);
+        }
+    }
+
+    void LineBetweenPositions(Vector3 start, Vector3 end)
+    {
+        GameObject lineObject = new GameObject("Line");
+        lineObject.tag = "LineTag";
+        LineRenderer line = lineObject.AddComponent(typeof(LineRenderer)) as LineRenderer;
+        line.positionCount = 2;
+        line.SetWidth(0.2f, 0.2f);
+        Vector3[] points = new Vector3[2];
+        points[0] = start;
+        points[1] = end;
+        line.SetPositions(points);
+    }
+
+    private void SetMouseHover()
+    {
+        Tile t = HoverTile;
+        while (t.parent)
+        {
+            LineBetweenPositions(t.transform.position, t.parent.transform.position);
+            t = t.parent;
+        }
+    }
+
     private void CheckMouseClick()
     {
         if (Input.GetMouseButtonUp(0))
         {
-            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
-
-            RaycastHit hit;
-            // Ignore a click on empty space
-            if (!Physics.Raycast(ray, out hit)) return;
-            
-            if (hit.collider.tag == "Tile")
+            Tile clickedTile = GetMouseTile();
+            if (clickedTile == null || !clickedTile.isSelectable) return;
+            ClearMouseHover();
+            if (clickedTile.occupant != null)
+            {
+                Action atk = GetComponent<ActionBasicAttack>();
+                atk.BeginAction(clickedTile);
+            }
+            else
+            {
+                Action move = GetComponent<ActionMove>();
+                move.BeginAction(clickedTile);
+            }
+        }
+        else
+        {
+            if (GetMouseTile() != HoverTile)
             {
-                Tile clickedTile = hit.collider.GetComponent<Tile>();
-
-                if (!clickedTile.isSelectable) return;
-                if (clickedTile.occupant != null)
-                {
-                    Action atk = GetComponent<ActionBasicAttack>();
-                    atk.BeginAction(clickedTile);
-                }
-                else
-                {
-                    Action move = GetComponent<ActionMove>();
-                    move.BeginAction(clickedTile);
-                }
+                ClearMouseHover();
+                HoverTile = GetMouseTile();
+                if (HoverTile == null) return;
```

Comment:
Maybe this belongs in SetMouseHover(), in case that method is ever called elsewhere.